app/src/main/java/fr/neamar/kiss/loader/LoadAppPojos.java

```diff
@@ -63,7 +68,18 @@ protected List<AppPojo> doInBackground(Void... params) {
                     ApplicationInfo appInfo = activityInfo.getApplicationInfo();
                     boolean disabled = PackageManagerUtils.isAppSuspended(appInfo) || isQuietModeEnabled(manager, profile);
                     final AppPojo app = createPojo(user, appInfo.packageName, activityInfo.getName(), activityInfo.getLabel(), disabled, excludedAppList, excludedFromHistoryAppList, excludedShortcutsAppList);
-                    apps.add(app);
+                    if ((android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM)
```

Comment:
Can you try to move code related to private space to separate class?

Code in `MainActivity`, in `LoadShortcutsPojos` class and in this class seems all pretty similar, this should be generalized somewhere instead of duplicated when possible.